MAINTAINERS_GUIDE.md

```diff
@@ -0,0 +1,131 @@
+from the great. So please be proud of your work, even the unglamorous
+parts, and encourage a culture of appreciation and respect for *every*
+aspect of improving the project -- not just the hot new features. 
+
+Being a maintainer is a time consuming commitment and should not be
+taken lightly. This document is a manual for maintainers old and new.
+It explains what is expected of maintainers, how they should work, and 
+what tools are available to them.
+
+This is a living document - if you see something out of date or missing,
+speak up!
+
+## What are a maintainer's responsibility?
+
+Part of a healthy project is to have active maintainers to support the
+community in contributions and perform tasks to keep the project running.
+It is every maintainer's responsibility to:
+
+* Keep the community a friendly place
+* Deliver prompt feedback and decisions on pull requests and mailing
+  list threads
+* Encourage other members to help each other in cases the maintainer
+  is overloaded or feels the lack of needed expertise
+* Make sure the changes made respects the philosophy, design and
+  roadmap of the project
+
+## How are decisions made?
+
+CRIU is an open-source project with an open design philosophy. This
+means that the repository is the source of truth for EVERY aspect of the
+project. *If it's part of the project, it's in the repo. It's in the 
+repo, it's part of the project.*
+
+All decisions affecting CRIU, big and small, follow the same 3 steps:
+
+  * Submit a change. Anyone can do this.
+
+  * Discuss it. Anyone can do this.
+
+  * Accept or decline it. Only maintainers do this
+
+*I'm a maintainer, should I make pull requests / send patches too?*
+
+Yes. Nobody should ever push to the repository directly. All changes
+should be made through submitting (and accepting) the change.
+
+### Two-steps decision making ###
+
+Since CRIU is extremely complex piece of software we try double hard
+not to make mistakes, that would be hard to fix in the future. In order
+to facilitate this, the "final" decision is made in two stages:
+
+  * We definitely want to try something out
+
+  * We think that the attempt was successful
+
+Technically, the former means "the patch is accepted into criu-dev branch",
+while the latter -- "the patch is accepted into the master branch".
+
+## Who decides what?
+
+All decisions are patches (either in form of pull requests, or sent to 
+the mailing list), and maintainers make decisions by merging or rejecting
+them. Review and approval or disagreement can be done by anyone and is
+denoted by adding a respective comment in the pull request. However,
+merging the change into either branch only happens after approvals from
+maintainers.
+
+In order for a patch to be merged into the criu-dev branch at least two
+maintainers should accept it. In order for a patch to be merged into the
+master branch the majority of maintainers should decide that (then prepare
+a pull request, submit it, ...)
```

Comment:
This raises the question of what happens with the commits in the `criu-dev` branch that are not being merged into the master branch.

Reply:
They continue sitting in the criu-dev untill the "feature" they are a part of is ready (and they get merged some time later), or until maintainers decide to throw them out. Presumably this is what happening right now.

Reply:
I understand, I think we should include this information in the Maintainers Guide.